### ISSUE TITLE
ng serve / ng test preventing deletion of files on windows subsystem for linux

### ISSUE THREAD

**Issue by JMBattista** [angular/angular-cli#6707]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
Windows 10, inside the WSL

@angular/cli: 1.1.0
node: 6.10.3
os: linux x64
@angular/animations: 4.1.3
@angular/common: 4.1.3
@angular/compiler: 4.1.3
@angular/core: 4.1.3
@angular/forms: 4.1.3
@angular/http: 4.1.3
@angular/platform-browser: 4.1.3
@angular/platform-browser-dynamic: 4.1.3
@angular/router: 4.1.3
@angular/cli: 1.1.0
@angular/compiler-cli: 4.1.3
@angular/language-service: 4.1.3

### Repro steps.

1. Create a new project with the CLI
2. Run "ng serve" or "ng test"
3. Edit a file (everything works to this point)
4. git reset --hard HEAD
5. The file has not been deleted correctly and cannot be deleted until ng serve/test are stopped.

### The log given by the failure.
angular cli does not return an error,.

### Desired functionality.
I should be able to delete files without having to cancel and restart ng serve / test

### Mention any other details that might be useful.
This is not exclusive to git reset commands. Any attempt to delete the files from command line, via the windows explorer, via an IDE, ect will all result in the same issues.

**Comment by Brocco** [comment]
Can you please explain what file(s) you're deleting as well as the output that you're seeing/not seeing?

**Comment by JMBattista** [comment]
Sure. This appears to apply to files generated by ng g, as well as temp files created by tools (such as IntelliJ). This specific example is around the files generated by ng g.
(have some project setup and run ng serve/test running)

    ng g c -it -is test
>  

    installing component
    create src/app/test/test.component.spec.ts
    create src/app/test/test.component.ts
    update src/app/app.module.ts
>
    git status
>

    On branch foo
    Changes not staged for commit:
      (use "git add <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   src/app/app.module.ts

    Untracked files:
      (use "git add <file>..." to include in what will be committed)

        src/app/test/

    no changes added to commit (use "git add" and/or "git commit -a")
>
    git reset --hard HEAD
>
    error: unable to create file src/app/app.module.ts (No such file or directory)
    fatal: Could not reset index file to revision 'HEAD'.
>
    git status
>
    On branch foo
    Changes not staged for commit:
      (use "git add/rm <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)

            deleted:    src/app/app.module.ts

    Untracked files:
      (use "git add <file>..." to include in what will be committed)

          src/app/test/

    no changes added to commit (use "git add" and/or "git commit -a")


If you cancel the ng serve/test command you instead get the following. Where git was able to successfully revert the app.module.ts file.

     git reset --hard HEAD
>

     HEAD is now at bcaefcf bar
>
    git status
>

    On branch foo
    Untracked files:
      (use "git add <file>..." to include in what will be committed)

            src/app/test/

    nothing added to commit but untracked files present (use "git add" to track)
>

Similarly if we try to directly delete the created src/app/test folder it files

    rm -R src/app/test
>

    rm: cannot remove ‘src/app/test/test.component.spec.ts’: No such file or directory
    rm: cannot remove ‘src/app/test/test.component.ts’: No such file or directory
    rm: cannot remove ‘src/app/test’: Directory not empty
>

    git status
> 

    On branch foo
    Changes not staged for commit:
      (use "git add <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)

          modified:   src/app/app.module.ts

    Untracked files:
      (use "git add <file>..." to include in what will be committed)

          src/app/test/

    no changes added to commit (use "git add" and/or "git commit -a")
>

If we disable ng serve/test before running the command we instead get

    rm -R src/app/test
    git status
>

    On branch foo
    Changes not staged for commit:
      (use "git add <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)

          modified:   src/app/app.module.ts

    no changes added to commit (use "git add" and/or "git commit -a")

**Comment by nicholasguyett** [comment]
Also on WSL,

```
@angular/cli: 1.1.2
node: 7.10.0
os: linux x64
@angular/animations: 4.2.3
@angular/common: 4.2.3
@angular/compiler: 4.2.3
@angular/core: 4.2.3
@angular/forms: 4.2.3
@angular/http: 4.2.3
@angular/platform-browser: 4.2.3
@angular/platform-browser-dynamic: 4.2.3
@angular/router: 4.2.3
@angular/cli: 1.1.2
@angular/compiler-cli: 4.2.3
@angular/language-service: 4.2.3
```

I had just initialized a brand new angular project via `ng new my-cool-app` and had not made any changes to my code before running `ng serve`.  When I edited the pre-generated `src/app/app.component.html` in order to begin actually implemented my application, the file disappeared on save.

Strangely enough, calling `ls` from a non-sudo context would display the errror `ls: cannot access 'src/app/app.component.html': No such file or directory` even when only calling it against the containing folder.  Using `sudo ls` does not throw any errors.  So it appears its the file itself isn't "disappearing", but rather the access to it is.  Attempting to edit via `sudo vim` instead of `vim` creates the same conditions.
